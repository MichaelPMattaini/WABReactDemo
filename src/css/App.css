.my-react-widget-back {
  background-color: rgb(209, 205, 205);
}
