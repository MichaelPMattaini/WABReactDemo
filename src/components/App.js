import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ErrorBoundary from './ErrorBoundary';
import MyComponent from './MyComponent';

import '../css/App.css';

import { sentry } from '../utils';

const propTypes = {
  wab: PropTypes.object,
  esriJS: PropTypes.object,
};

const defaultProps = {
  wab: {},
  esriJS: {},
};

export default class App extends Component {
  state = {
    loading: true,
  };

  componentDidMount() {
    console.info('Sentry is initialized: ', sentry.isSetup());
  }

  render() {
    return (
      <div>
        <ErrorBoundary>
          <MyComponent
            title="WAB Widget"
            wab={this.props.wab}
            esriJS={this.props.esriJS}
          />
        </ErrorBoundary>
      </div>
    );
  }
}

App.propTypes = propTypes;
App.defaultProps = defaultProps;
