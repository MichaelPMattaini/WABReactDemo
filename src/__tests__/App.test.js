import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../components/App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('displays entry to widget', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
