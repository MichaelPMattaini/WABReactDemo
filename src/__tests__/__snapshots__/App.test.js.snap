// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> displays entry to widget 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
